### ISSUE TITLE
Assert on exit with winit 0.30

### ISSUE THREAD

**Issue by waywardmonkeys** [linebender/xilem#239]
When I run the `hello_xilem` or `mason` examples from 30c838b4004c999225194f9521c7999e19e3017b, I get this error when closing the window:

```
thread 'main' panicked at ~/.cargo/registry/src/index.crates.io-6f17d22bba15001f/winit-0.30.0/src/platform_impl/macos/app_delegate.rs:147:39:
a delegate was not configured on the application
stack backtrace:
   0: rust_begin_unwind
             at /rustc/25ef9e3d85d934b27d9dada2f9dd52b1dc63bb04/library/std/src/panicking.rs:647:5
   1: core::panicking::panic_fmt
             at /rustc/25ef9e3d85d934b27d9dada2f9dd52b1dc63bb04/library/core/src/panicking.rs:72:14
   2: core::panicking::panic_display
             at /rustc/25ef9e3d85d934b27d9dada2f9dd52b1dc63bb04/library/core/src/panicking.rs:196:5
   3: core::panicking::panic_str
             at /rustc/25ef9e3d85d934b27d9dada2f9dd52b1dc63bb04/library/core/src/panicking.rs:171:5
   4: core::option::expect_failed
             at /rustc/25ef9e3d85d934b27d9dada2f9dd52b1dc63bb04/library/core/src/option.rs:1988:5
   5: core::option::Option<T>::expect
             at /rustc/25ef9e3d85d934b27d9dada2f9dd52b1dc63bb04/library/core/src/option.rs:894:21
   6: winit::platform_impl::platform::app_delegate::ApplicationDelegate::get
             at ~/.cargo/registry/src/index.crates.io-6f17d22bba15001f/winit-0.30.0/src/platform_impl/macos/app_delegate.rs:147:13
   7: winit::platform_impl::platform::window_delegate::WindowDelegate::queue_event
             at ~/.cargo/registry/src/index.crates.io-6f17d22bba15001f/winit-0.30.0/src/platform_impl/macos/window_delegate.rs:759:28
   8: winit::platform_impl::platform::window_delegate::WindowDelegate::window_will_close
             at ~/.cargo/registry/src/index.crates.io-6f17d22bba15001f/winit-0.30.0/src/platform_impl/macos/window_delegate.rs:152:13
   9: <unknown>
  10: <unknown>
  11: <unknown>
  12: <unknown>
  13: <unknown>
  14: <unknown>
  15: <unknown>
  16: <() as objc2::encode::EncodeArguments>::__invoke
             at ~/.cargo/registry/src/index.crates.io-6f17d22bba15001f/objc2-0.5.1/src/encode.rs:437:26
  17: objc2::runtime::message_receiver::msg_send_primitive::send
             at ~/.cargo/registry/src/index.crates.io-6f17d22bba15001f/objc2-0.5.1/src/runtime/message_receiver.rs:173:18
  18: objc2::runtime::message_receiver::MessageReceiver::send_message
             at ~/.cargo/registry/src/index.crates.io-6f17d22bba15001f/objc2-0.5.1/src/runtime/message_receiver.rs:408:38
  19: objc2::__macro_helpers::msg_send::MsgSend::send_message
             at ~/.cargo/registry/src/index.crates.io-6f17d22bba15001f/objc2-0.5.1/src/__macro_helpers/msg_send.rs:27:31
  20: objc2_app_kit::generated::__NSWindow::NSWindow::close
             at ~/.cargo/registry/src/index.crates.io-6f17d22bba15001f/objc2-0.5.1/src/macros/extern_methods.rs:272:14
  21: <winit::platform_impl::platform::window::Window as core::ops::drop::Drop>::drop::{{closure}}::{{closure}}
             at ~/.cargo/registry/src/index.crates.io-6f17d22bba15001f/winit-0.30.0/src/platform_impl/macos/window.rs:21:62
  22: objc2::rc::autorelease::autoreleasepool
             at ~/.cargo/registry/src/index.crates.io-6f17d22bba15001f/objc2-0.5.1/src/rc/autorelease.rs:438:15
  23: <winit::platform_impl::platform::window::Window as core::ops::drop::Drop>::drop::{{closure}}
             at ~/.cargo/registry/src/index.crates.io-6f17d22bba15001f/winit-0.30.0/src/platform_impl/macos/window.rs:21:42
  24: objc2_foundation::thread::MainThreadBound<T>::get_on_main::{{closure}}
             at ~/.cargo/registry/src/index.crates.io-6f17d22bba15001f/objc2-foundation-0.2.0/src/thread.rs:411:27
  25: objc2_foundation::thread::run_on_main
             at ~/.cargo/registry/src/index.crates.io-6f17d22bba15001f/objc2-foundation-0.2.0/src/thread.rs:84:9
  26: objc2_foundation::thread::MainThreadBound<T>::get_on_main
             at ~/.cargo/registry/src/index.crates.io-6f17d22bba15001f/objc2-foundation-0.2.0/src/thread.rs:411:9
  27: <winit::platform_impl::platform::window::Window as core::ops::drop::Drop>::drop
             at ~/.cargo/registry/src/index.crates.io-6f17d22bba15001f/winit-0.30.0/src/platform_impl/macos/window.rs:21:9
  28: core::ptr::drop_in_place<winit::platform_impl::platform::window::Window>
             at /rustc/25ef9e3d85d934b27d9dada2f9dd52b1dc63bb04/library/core/src/ptr/mod.rs:507:1
  29: core::ptr::drop_in_place<winit::window::Window>
             at /rustc/25ef9e3d85d934b27d9dada2f9dd52b1dc63bb04/library/core/src/ptr/mod.rs:507:1
  30: alloc::sync::Arc<T,A>::drop_slow
             at /rustc/25ef9e3d85d934b27d9dada2f9dd52b1dc63bb04/library/alloc/src/sync.rs:1756:18
  31: <alloc::sync::Arc<T,A> as core::ops::drop::Drop>::drop
             at /rustc/25ef9e3d85d934b27d9dada2f9dd52b1dc63bb04/library/alloc/src/sync.rs:2411:13
  32: core::ptr::drop_in_place<alloc::sync::Arc<winit::window::Window>>
             at /rustc/25ef9e3d85d934b27d9dada2f9dd52b1dc63bb04/library/core/src/ptr/mod.rs:507:1
  33: core::ptr::drop_in_place<alloc::boxed::Box<dyn wgpu::WindowHandle>>
             at /rustc/25ef9e3d85d934b27d9dada2f9dd52b1dc63bb04/library/core/src/ptr/mod.rs:507:1
  34: core::ptr::drop_in_place<core::option::Option<alloc::boxed::Box<dyn wgpu::WindowHandle>>>
             at /rustc/25ef9e3d85d934b27d9dada2f9dd52b1dc63bb04/library/core/src/ptr/mod.rs:507:1
  35: core::ptr::drop_in_place<wgpu::Surface>
             at /rustc/25ef9e3d85d934b27d9dada2f9dd52b1dc63bb04/library/core/src/ptr/mod.rs:507:1
  36: core::ptr::drop_in_place<vello::util::RenderSurface>
             at /rustc/25ef9e3d85d934b27d9dada2f9dd52b1dc63bb04/library/core/src/ptr/mod.rs:507:1
  37: core::ptr::drop_in_place<masonry::event_loop_runner::MainState>
             at /rustc/25ef9e3d85d934b27d9dada2f9dd52b1dc63bb04/library/core/src/ptr/mod.rs:507:1
  38: masonry::event_loop_runner::run
             at .../xilem/crates/masonry/src/event_loop_runner.rs:56:1
  39: xilem_masonry::Xilem<State,Logic,View>::run_windowed_in
             at ./src/lib.rs:199:9
  40: xilem_masonry::Xilem<State,Logic,View>::run_windowed
             at ./src/lib.rs:185:9
  41: mason::main
             at ./examples/mason.rs:77:5
  42: core::ops::function::FnOnce::call_once
             at /rustc/25ef9e3d85d934b27d9dada2f9dd52b1dc63bb04/library/core/src/ops/function.rs:250:5
note: Some details are omitted, run with `RUST_BACKTRACE=full` for a verbose backtrace.
```

FYI: @xorgy

**Comment by xorgy** [comment]
I don't have a Mac with me, I wasn't seeing the same when I ran them. I could take a look though.

**Comment by waywardmonkeys** [comment]
I think this was fixed in winit 0.30.1 via this: https://github.com/rust-windowing/winit/commit/d24be6478198b94083da42e2f3e53d3ba479e7d0
